README.md

```diff
@@ -0,0 +1,206 @@
+# arm-builder-kernels
+
+## ToC
+* [Ubuntu](#ubuntu)
+* [Amazon Linux 2](#amazon-linux-2)
+
+## Ubuntu
+* update `build/kernel-src` submodule and checkout new upstream version tag
+* generate `patch/{{version-tag}}/balena.patch` from existing/previous patch(es)
+* use pull-request workflow to create a GitHub release containing Debian kernel binary (.deb)
+
+
+## Amazon Linux 2
+
+### install dependencies
+> use next generation Linux kernel as a base on a temporary AWS EC2 build instance (aarch64)
+
+    amazon-linux-extras install kernel-ng
+
+    yum groupinstall -y "Development Tools"
+
+    yum-builddep -y kernel && yum install -y pesign
+
+    yumdownloader --source kernel
+
+    groupadd mock && useradd mockbuild
+
+
+### unpack RPM
+> https://www.hiroom2.com/2016/05/29/centos-7-rebuild-kernel-with-src-rpm/
+
+    rpm -i kernel-*.src.rpm
+
+    pushd rpmbuild
+
+    cp -a SPECS/kernel.spec{,.org}
+
+
+### increment build version
+> increment build number in `SPECS/kernel.spec`
+
+    diff -uprN SPECS/kernel.spec{.org,}
+
+
+### prepare kernel source
+
+    rpmbuild -bp SPECS/kernel.spec
+
+    pushd BUILD/kernel-*
+
+    kernel_tree=$(find . -maxdepth 1 -name linux-*.amzn2.aarch64 -type d)
+
+    cp -a ${kernel_tree}{,.org}
+
+
+### patch
```

Comment:
This patch may not be necessary with the Ubuntu kernel sources. See [this patch](https://github.com/balena-os/balena-generic/blob/master/layers/meta-balena-generic/recipes-kernel/linux/files/generic-aarch64/0001-UBUNTU-SAUCE-no-up-add-compat_uts_machine-kernel-com.patch) applied in balenaOS from the Ubuntu kernel. It adds the `compat_uts_machine` kernel cmdline parameter that allows for overriding `uts_machine` with an arbitrary string at boot.